src/logging.rs

```diff
@@ -0,0 +1,69 @@
+//! Logging related utilities.
+
+use log::{self, Log, Record, Level, Metadata, SetLoggerError};
+
+use super::Sentry;
+
+/// Logger which implements the `log::Log` trait. This allows logging via the
+/// macros defined in the `log` crate.
+pub struct SentryLogger {
+  // Sentry client used for delivering log messages.
+  sentry: Sentry,
+
+  // Name of the logger to log messages with.
+  logger_name: String,
+
+  // Minimum level to log messages to Sentry at.
+  level: Level,
+}
+
+impl SentryLogger {
+  /// Construct a new `SentryLogger`.
+  ///
+  /// # Arguments
+  ///
+  /// * `sentry` - Sentry client used to deliver log messages.
+  /// * `logger_name` - String used as logger name in messages.
+  /// * `level` - Minimum level to log messages to Sentry at.
+  pub fn new(sentry: Sentry, logger_name: &str, level: Level) -> Self {
```

Comment:
I know this adds more work for the example, but I'm more of the idea we should take logger_name as a String here (instead of as a str).

The reason for this is because we're hiding a clone behind them, and it *always clones* with this implementation. Even if I already have a String. Say for example I have:

```rust
let a_str: String = get_some_cool_string_from_somewhere!();
// I now have to "clone" a_str in order to create a logger even though I already have an instance of "String".
SentryLogger::new(sentry, &a_str, level);
```

Thus taking a `String` here (and not using to_owned()) allows us to not "hide" anything away. It's very explicit we need ownership, and we won't clone or anything under the hood.

Reply:
Good point, will also fix that here.